include/swift/AST/DiagnosticsCommon.def

```diff
@@ -95,6 +95,10 @@ NOTE(profile_read_error,none,
 ERROR(generic_signature_not_minimal,none,
       "generic requirement '%0' is redundant in %1", (StringRef, StringRef))
 
+WARNING(protocol_extension_redundant_requirement,none,
```

Comment:
How about `requirement of '%1' to '%2' is redundant in an extension of '%0'`?

Reply:
Definitely a lot more succinct! I have updated the diagnostic message now.